src/main/java/com/github/manolo8/darkbot/core/manager/GuiManager.java

```diff
@@ -208,7 +213,17 @@ public boolean canTickModule() {
         }
 
 
+
         HeroManager hero = main.hero;
+        if (main.config.MISCELLANEOUS.REFRESH_AFTER_REVIVE
+                && this.needRefresh
+                && System.currentTimeMillis() - lastRepair > 5_000
+                && hero.health.hp >= 1_000 ){
+            System.out.println("Triggering refresh: refreshing after death");
+            API.handleRefresh();
+            this.needRefresh = false;
+            return false;
+        }
```

Comment:
If you keep the logic this way, needRefresh will stay in true forever if you have your setting set to false, and the instant you enable it it will trigger a refresh.

Make it so the setting is checked inside a nested if, and it affects if the log, the api call, and the return, but still allows needsRefresh to be set back to false after the time passed